### PR TITLE
FAI-6714 - Convert undefined source record property values to nulls

### PR DESCRIPTION
**Pull request by cjwooo**
## Description

> Provide description here

## Type of change
- [ ] Bug fix
- [ ] New feature
- [ ] Breaking change

## Related issues

> Fix [#1]() 

## Migration notes

> Describe migration notes if any

## Extra info

> Add any additional information

### CODE REVIEW

**Review by ted-faros — APPROVED** [review]
💥

File: faros-airbyte-cdk/test/logger.test.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+  test('converts undefined record fields to nulls', async () => {
+    const data = {
+      boolean: true,
+      number: 1,
+      string: 'string',
+      array: [1, 'string', {a: 1}],
+      null: null,
+      undefined: undefined,
+      object: {a: 1, b: 'string', null: null, undefined: undefined},
```
Comment (tovbinm) [inline faros-airbyte-cdk/test/logger.test.ts:36]: ehh, `null: null, undefined: undefined` is weird! we really only care about values for non-empty keys

File: faros-airbyte-cdk/src/logger.ts [hunk 2]
```diff
@@ -86,6 +87,13 @@ export class AirbyteLogger {
       const msgLevelOrder = AirbyteLogLevelOrder((msg as AirbyteLog).log.level);
       if (levelOrder > msgLevelOrder) return;
     }
-    console.log(JSON.stringify(msg));
+
+    // convert undefined record values to nulls so they are stringified
+    const replacer =
+      msg.type === AirbyteMessageType.RECORD
```
Comment (tovbinm) [inline faros-airbyte-cdk/src/logger.ts:93]: we can also skip **keys**  which are empty, null or undefined

**Review by tovbinm — APPROVED** [review]
lgtm, see comments

File: faros-airbyte-cdk/test/logger.test.ts (hunk 1 above)
Comment (cjwooo) [reply faros-airbyte-cdk/test/logger.test.ts:36]: So it's ok to have a record field value be an object with undefined values? They will not be passed to the destination.

File: faros-airbyte-cdk/src/logger.ts (hunk 2 above)
Comment (cjwooo) [reply faros-airbyte-cdk/src/logger.ts:93]: From ChatGPT:

No, it is not possible for a JavaScript object to have a null or undefined key name. In JavaScript, object keys must be of type string or symbol.

If you attempt to use null or undefined as a key name when creating an object, they will be implicitly converted to strings.

Here's an example:

```
const obj = {};

obj[null] = 'Value for null key';
obj[undefined] = 'Value for undefined key';

console.log(obj);
```

In this example, when assigning values to null and undefined key names, JavaScript implicitly converts them to strings. The resulting object will have string keys named "null" and "undefined".

The output of console.log(obj) will be:
```
{ null: 'Value for null key', undefined: 'Value for undefined key' }
```
As you can see, the null and undefined keys are treated as strings in the resulting object.

Comment (ted-faros) [reply faros-airbyte-cdk/src/logger.ts:93]: no, please don't filter the key.  I like the PR as-is.

File: faros-airbyte-cdk/test/logger.test.ts (hunk 1 above)
Comment (cjwooo) [reply faros-airbyte-cdk/test/logger.test.ts:36]: Actually I think you're misinterpreting. The **keys** in `{null: null, undefined: undefined}`, are the strings `"null"` and `"undefined"` respectively. The **values** are `null` and `undefined`.

Comment (cjwooo) [reply faros-airbyte-cdk/test/logger.test.ts]: I've updated the test to make this clearer.

File: faros-airbyte-cdk/src/logger.ts (hunk 2 above)
Comment (tovbinm) [reply faros-airbyte-cdk/src/logger.ts:93]: Let's just fix the tests at least. Null / undefined keys are confusing.

Comment (tovbinm) [reply faros-airbyte-cdk/src/logger.ts:93]: You did it already. Thanks.

**Review by tovbinm — APPROVED** [review]

**Pull request merged**